### ISSUE TITLE
SLIC: Populate rank and count earlier

### ISSUE THREAD

**Issue by white238** [LLNL/axom#1410]
From @gberg617:

I was noticing that the rank information (both list of ranks and count) are populated for Lumberjack streams only when pushing the messages.  Do you know if it would be possible at all to populate this information for ranks that are going call outputLocal and then abort?  No need to go out of your way if you think this will be difficult to change, but I was curious if it's something that could be a 
simple few lines of code changes.

**Comment by bmhan12** [comment]
@gberg617 
My initial thought was I expected the rank information to be present for `outputLocal`, since we are testing for it in our unit tests. 

Looks like we are not checking the rank information yet in our [unit tests](https://github.com/LLNL/axom/blob/efef4ec471e90cde65c6059eb0e6094c50430981/src/axom/slic/tests/slic_macros.cpp#L50-L115).

I'll add unit tests, and see if I get the same missing-rank-information behavior.

**Comment by bmhan12** [comment]
@gberg617 

So I am encountering errors in terms of unexpected rank/rank count information when I added unit tests (branch: [bugfix/han12/slic_rank_info](https://github.com/LLNL/axom/tree/bugfix/han12/slic_rank_info))
Still in the process of debugging what/where the issue is from...

**Comment by gberg617** [comment]
> @gberg617
> 
> So I am encountering errors in terms of unexpected rank/rank count information when I added unit tests (branch: [bugfix/han12/slic_rank_info](https://github.com/LLNL/axom/tree/bugfix/han12/slic_rank_info)) Still in the process of debugging what/where the issue is from...

Thanks for the update!

**Comment by bmhan12** [comment]
@gberg617 
I added rank and rank count unit tests in PR #1431.
The error I was running into was strictly Axom-related, involving `std::ends` (PR has more details).

In terms of rank and rank count, I did not notice the information missing on `outputLocalMessages()` calls.

I did want to iterate that for `outputLocalMessages()` the messages do not get combined, as that is a collective operation that does not bode well when trying to `MPI_Abort()`.
So its expected that the messages will only contain information for the current rank (e.g. Rank 0's message will have `rank: 0 rank_count 1` , Rank 1's message will have `rank: 1 rank_count: 1`, etc.)
